server_config_test.cc

```diff
@@ -0,0 +1,61 @@
+#include "config_parser.h"
+#include "gtest/gtest.h"
+#include "request_handlers.h"
+#include "server_config.h"
+
+#include <string>
+#include <sstream>
+
+class ServerConfigTest : public ::testing::Test {
+protected:
+  // Helper method to pass arbitrary strings into the parser
+  std::unique_ptr<ServerConfig> ParseString(const std::string config_string) {
+    std::stringstream config_stream(config_string);
+    parser_.Parse(&config_stream, &out_config_);
+
+    return std::unique_ptr<ServerConfig>(new ServerConfig(out_config_));
+  }
+
+  NginxConfigParser parser_;
+  NginxConfig out_config_;
+};
+
+TEST_F(ServerConfigTest, NoPortConfig) {
+    std::unique_ptr<ServerConfig> server_config = ParseString("server { ACM Hack; }");
+    EXPECT_EQ("", server_config->propertyLookUp("port"));
```

Comment:
Nice catch, I forgot to include fail-case messages. I've added them in https://github.com/UCLA-CS130/Mr.-Robot-et-al./commit/a793e20edc43ce7bd2b33372a4fee553851709c6